text/0033-fake-account-packet/0033-fake-account-packet.md

```diff
@@ -0,0 +1,206 @@
+- Feature Name: Fake Account Packet
+- Status: proposed
+- Type: enhancement
+- Related components: SAFE Vault and Routing
+- Start Date: 10-05-2016
+- RFC PR:
+- Issue number:
+
+# Summary
+
+This proposes an enhancement whereby the network will respond to a `Get` request for a non-existent
+account packet (`StructuredData` with `type_tag` 0) with a fake account packet.
+
+# Motivation
+
+The objective is to deter an attack where random account packets are requested by a malicious Client
+in the hopes of finding one in order to attempt decryption of it.
+
+The defence is to have the network respond to requests for non-existent account packets with fakes
+which will appear valid to the recipient.  This will make it expensive for the attacker, since it
+won't know whether the decrypted packet will yield useful data or not.
+
+# Detailed design
+
+## Issues to Overcome
+
+The main issue is to be able to have a group of Vaults agree a fake packet which will be accumulated
+by the attacker - i.e. the group all need to generate an identical fake.  Essentially we just need a
+single value agreed across the group which can then be used to seed a pseudorandom number generator.
+
+The Vaults could all exchange a message in order to agree a seed, however this is undesirable since
+it could easily be exploited as an amplification attack.
+
+This leaves little scope for agreeing a single seed; the only piece of information which all the
+Vaults know is the list of Vault names comprising the close group.  This is useless as a seed since
+the attacker could simply monitor the individual Routing responses arriving and hence accrue this
+list of Vault names.  It would then be easy to deduce that the responses contain a fake.
+
+A more trivial issue is that while account packets are currently the highest-value targets for an
+attacker, it should be possible to extend the defence to other `StructuredData` types easily.
+
+## Proposed Solution
+
+There is an example implementation at https://gitlab.com/Fraser999/Fake-Account-Packet.
+
+The implementation here requires Vaults to hold a new piece of data; a [`SessionId`]
+(https://gitlab.com/Fraser999/Fake-Account-Packet/blob/master/src/main.rs#L21) defined as
+
+```rust
+struct SessionId(pub [u8; SEEDBYTES]);
+```
+
+where `SEEDBYTES` is 32; the size of the sodiumoxide `Seed`.
+
+This `SessionId` is created randomly by the Vault on startup and is exchanged with any peers with
+which it establishes a connection.  Critically, this `SessionId` is never sent to any node (Vault or
+Client) until the peer is being added to the Routing Table.  This should reduce the ability for an
+attacker to harvest `SessionId`s.
+
+This new piece of information per Vault (unknown by Clients) can now be used to generate a seed for
+any given close group.  To avoid generating the same seed for all addressable elements within a
+narrow address space it would be best to also mix in the requested data name.
```

Comment:
I'm not sure I follow.  The seeds aren't transmitted to the client - there's no feasible way for the attacker to know what seeds were used as far as I can see.

I don't think we benefit here using hashing - but maybe I'm missing something?